static/js/compose_ui.js

```diff
@@ -67,11 +67,39 @@ export function smart_insert($textarea, syntax) {
     autosize_textarea($textarea);
 }
 
-export function insert_syntax_and_focus(syntax, $textarea = $("#compose-textarea")) {
+export function smart_insert_block($textarea, syntax) {
+    function is_newline(c) {
+        return c === "\n" || c === "\r";
+    }
```

Comment:
It’s not possible to have `\r` in a `<textarea>`; browsers [normalize](https://html.spec.whatwg.org/multipage/form-elements.html#the-textarea-element) to `\n` only.

> When a [`textarea`](https://html.spec.whatwg.org/multipage/form-elements.html#the-textarea-element) is [mutable](https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#concept-fe-mutable), its [raw value](https://html.spec.whatwg.org/multipage/form-elements.html#concept-textarea-raw-value) should be editable by the user: the user agent should allow the user to edit, insert, and remove text, and to insert and remove line breaks in the form of U+000A LINE FEED (LF) characters.

Reply:
This is great to know, thank you! I think I'll just remove the `is_newline()` function and put the logic elsewhere due to the simplicity of the check in that case.